modules/FindImGui.cmake

```diff
@@ -0,0 +1,130 @@
+#   the rights to use, copy, modify, merge, publish, distribute, sublicense,
+#   and/or sell copies of the Software, and to permit persons to whom the
+#   Software is furnished to do so, subject to the following conditions:
+#
+#   The above copyright notice and this permission notice shall be included
+#   in all copies or substantial portions of the Software.
+#
+#   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+#   THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#   DEALINGS IN THE SOFTWARE.
+#
+
+find_package(imgui CONFIG QUIET)
+if(imgui_FOUND)
+    if(NOT TARGET ImGui::ImGui)
+        add_library(ImGui::ImGui INTERFACE IMPORTED)
+        set_property(TARGET ImGui::ImGui APPEND PROPERTY
+            INTERFACE_LINK_LIBRARIES imgui::imgui)
+
+        # Retrieve include directory for FindPackageHandleStandardArgs later
+        get_target_property(ImGui_INCLUDE_DIR imgui::imgui
+            INTERFACE_INCLUDE_DIRECTORIES)
+
+        add_library(ImGui::Sources INTERFACE IMPORTED)
+        set_property(TARGET ImGui::Sources APPEND PROPERTY
+            INTERFACE_LINK_LIBRARIES ImGui::ImGui)
+    endif()
+else()
+    # Disable the find root path here, it overrides the
+    # CMAKE_FIND_ROOT_PATH_MODE_INCLUDE setting potentially set in
+    # toolchains.
+    find_path(ImGui_INCLUDE_DIR NAMES imgui.h HINTS "${IMGUI_DIR}"
+        NO_CMAKE_FIND_ROOT_PATH)
+
+    if(NOT TARGET ImGui::ImGui)
+        add_library(ImGui::ImGui INTERFACE IMPORTED)
+        set_property(TARGET ImGui::ImGui APPEND PROPERTY
+            INTERFACE_INCLUDE_DIRECTORIES "${ImGui_INCLUDE_DIR}")
+    endif()
+endif()
+
+# Find components
+foreach(_component IN LISTS ImGui_FIND_COMPONENTS)
+
+    if(_component STREQUAL "Sources")
+        set(ImGui_Sources_FOUND TRUE)
+        set(ImGui_SOURCES )
+
+        foreach(_file imgui imgui_widgets imgui_draw imgui_demo)
+            # Disable the find root path here, it overrides the
+            # CMAKE_FIND_ROOT_PATH_MODE_INCLUDE setting potentially set in
+            # toolchains.
+            find_file(ImGui_${_file}_SOURCE NAMES ${_file}.cpp
+                HINTS "${IMGUI_DIR}" NO_CMAKE_FIND_ROOT_PATH)
+            list(APPEND ImGui_SOURCES ${ImGui_${_file}_SOURCE})
+
+            if(NOT ImGui_${_file}_SOURCE)
+                set(ImGui_Sources_FOUND FALSE)
+                continue()
+            endif()
+
+            # Hide warnings from imgui source files
+
+            set_property(SOURCE ${ImGui_${_file}_SOURCE} APPEND PROPERTY COMPILE_DEFINITIONS
+                "IMGUI_USER_CONFIG=\"Magnum/ImGuiIntegration/visibility.h\"")
+
+            # Clang-specific flags
```

Comment:
`GCC- and Clang-specific` (sorry, my snippet was wrong). Also, maybe above a comment on what the IMGUI_USER_CONFIG does?